hpycc/utils/parsers.py

```diff
@@ -153,22 +149,43 @@ def parse_schema_from_xml(xml):
     xml : str
         xml string returned by ecl run. This is located in the json
         as ["WUResultResponse]["Result"]["XmlSchema"]["xml"].
+    dtype : None, dict
+        data types to be parsed from ecl. If None, these are inferred.
 
     Returns
     -------
+    OrderedDict
+        dict of column stats, in the form {name: Str, type: Str, is_a_set: Bool}.
     list
-        List of namedtuples in form [(name, is_set, type)].
+        Column names in order of occurrence.
 
     """
     x = xml.replace("\n", "")
     xml = ElementTree.fromstring(x)
     schema = xml[0][0][0][0][0][0]
 
-    return (Schema(
-        child.attrib["name"],
-        "type" not in child.keys(),
-        get_python_type_from_ecl_type(child)
-    ) for child in schema)
+    schema_out = OrderedDict()
+    for child in schema:
+        name = child.attrib["name"]
+        is_set = "type" not in child.keys()
+        typ = _determine_dtype(dtype, name, child)
+        schema_out[name] = {'type': typ, 'is_a_set': is_set}
+
+    # Check that no weird columns have been passed
+    if isinstance(dtype, dict) and any([dtype_col not in schema_out.keys() for dtype_col in dtype.keys()]):
+        raise KeyError('Not all dtype columns exist in the logical file!\nFound: %s\nGiven: %s' %
+                       (schema_out.keys(), dtype))
+
+    return schema_out
+
+
+def _determine_dtype(dtype, name, child):
```

Comment:
I dont get why we need this

Reply:
Popped out into a seperate function.

Reply:
and removed the stuff from schema parsing.